File: advent-of-code/kotlin/src/adventofcode/day01/Day01.kt
package adventofcode.day01

import adventofcode.AdventOfCodeSolution

fun main() {
    Day01.solve()
}

object Day01 : AdventOfCodeSolution<Int>() {
    override fun solve() {
        solve(1, 1, 1)
    }

    override fun part1(input: List<String>): Int {
        return input.size
    }

    override fun part2(input: List<String>): Int {
        return input.size
    }
}

File: advent-of-code/c++/src/adventofcode/day01/day01.cpp
#include <string>
#include <vector>

#include "../advent_of_code_solution.hpp"

class Day01 : public AdventOfCodeSolution<int, int> {
private:
    typedef AdventOfCodeSolution<int, int> super;
public:
    Day01() {
    }

    ~Day01() {
    }

    int part1(std::vector<std::string> data) {
        return data.size();
    }

    int part2(std::vector<std::string> data) {
        return data.size();
    }

    void solve() {
        super::solve(1, 1, 1);
    }
};

int main() {
    Day01 day;
    day.solve();
    return 0;
}